app/questionnaire/path_finder.py

```diff
@@ -59,19 +59,34 @@ def build_path(self):
                 group_blocks = self._build_blocks_for_group(group, instance_idx)
                 blocks += group_blocks
 
-            if group['id'] in first_groups:
-                this_location = Location(group['id'], 0, group['blocks'][0]['id'])
+            first_group_instance_index = self._get_first_group_instance_index_in_block(group, blocks)
+            all_blocks_skipped = first_group_instance_index is None
 
-            if blocks:
-                path, block_index = self._build_path_within_group(blocks, block_index, this_location, path)
+            if not all_blocks_skipped:
+                if group['id'] in first_groups:
+                    this_location = Location(group['id'], first_group_instance_index, first_block_in_group)
+
+                if blocks:
+                    path, block_index = self._build_path_within_group(blocks, block_index, this_location, path)
 
         return path
 
+    @staticmethod
+    def _get_first_group_instance_index_in_block(group, blocks):
+        # returns the first block of a list of blocks that matches the group id
+
+        try:
+            first_block = next(block for block in blocks if block['group_id'] == group['id'])
+            return first_block['group_instance']
+        except (StopIteration, KeyError):  # Either no group instances (all skipped) or group has no 'id'
+            pass
+        return None
```

Comment:
Pedantic: I wouldn't bother with this line

Reply:
I know , but I thought best to make it explicit